dhcpv4/client.go

```diff
@@ -44,69 +76,62 @@ func MakeRawBroadcastPacket(payload []byte) ([]byte, error) {
 	return ret, nil
 }
 
-// Run a full DORA transaction: Discovery, Offer, Request, Acknowledge, over
-// UDP. Does not retry in case of failures.
-// Returns a list of DHCPv4 structures representing the exchange. It can contain
-// up to four elements, ordered as Discovery, Offer, Request and Acknowledge.
-// In case of errors, an error is returned, and the list of DHCPv4 objects will
-// be shorted than 4, containing all the sent and received DHCPv4 messages.
-func (c *Client) Exchange(ifname string, d *DHCPv4) ([]DHCPv4, error) {
-	conversation := make([]DHCPv4, 1)
-	var err error
-
-	// Discovery
-	if d == nil {
-		d, err = NewDiscoveryForInterface(ifname)
-	}
-	conversation[0] = *d
-
+// MakeBroadcastSocket creates a socket that can be passed to syscall.Sendto
+// that will send packets out to the broadcast address.
+func MakeBroadcastSocket(ifname string) (int, error) {
 	fd, err := syscall.Socket(syscall.AF_INET, syscall.SOCK_RAW, syscall.IPPROTO_RAW)
 	if err != nil {
-		return conversation, err
+		return -1, err
```

Comment:
I would just `return fd, err` here and below, since the error indicator is already in `err`